cranelift/codegen/src/isa/x64/abi.rs

```diff
@@ -0,0 +1,457 @@
+                    ))
+                } else {
+                    unimplemented!("moving from vreg to non-int return value");
+                }
+            }
+
+            ABIRet::_Stack => {
+                unimplemented!("moving from vreg to stack return value");
+            }
+        }
+
+        ret
+    }
+
+    fn gen_ret(&self) -> Inst {
+        Inst::ret()
+    }
+
+    fn gen_epilogue_placeholder(&self) -> Inst {
+        Inst::epilogue_placeholder()
+    }
+
+    fn set_num_spillslots(&mut self, slots: usize) {
+        self.num_spill_slots = Some(slots);
+    }
+
+    fn set_clobbered(&mut self, clobbered: Set<Writable<RealReg>>) {
+        self.clobbered = clobbered;
+    }
+
+    fn stackslot_addr(&self, _slot: StackSlot, _offset: u32, _into_reg: Writable<Reg>) -> Inst {
+        unimplemented!()
+    }
+
+    fn load_stackslot(
+        &self,
+        _slot: StackSlot,
+        _offset: u32,
+        _ty: Type,
+        _into_reg: Writable<Reg>,
+    ) -> Inst {
+        unimplemented!("load_stackslot")
+    }
+
+    fn store_stackslot(&self, _slot: StackSlot, _offset: u32, _ty: Type, _from_reg: Reg) -> Inst {
+        unimplemented!("store_stackslot")
+    }
+
+    fn load_spillslot(&self, _slot: SpillSlot, _ty: Type, _into_reg: Writable<Reg>) -> Inst {
+        unimplemented!("load_spillslot")
+    }
+
+    fn store_spillslot(&self, _slot: SpillSlot, _ty: Type, _from_reg: Reg) -> Inst {
+        unimplemented!("store_spillslot")
+    }
+
+    fn gen_prologue(&mut self) -> Vec<Inst> {
+        let r_rsp = regs::rsp();
+
+        let mut insts = vec![];
+
+        // Baldrdash generates its own prologue sequence, so we don't have to.
+        if !self.call_conv.extends_baldrdash() {
+            let r_rbp = regs::rbp();
+            let w_rbp = Writable::<Reg>::from_reg(r_rbp);
+
+            // The "traditional" pre-preamble
+            // RSP before the call will be 0 % 16.  So here, it is 8 % 16.
+            insts.push(Inst::push64(RMI::reg(r_rbp)));
+            // RSP is now 0 % 16
+            insts.push(Inst::mov_r_r(true, r_rsp, w_rbp));
+        }
+
+        // Save callee saved registers that we trash. Keep track of how much space we've used, so
+        // as to know what we have to do to get the base of the spill area 0 % 16.
+        let mut callee_saved_used = 0;
+        let clobbered = get_callee_saves(self.clobbered.to_vec());
+        for reg in clobbered {
+            let r_reg = reg.to_reg();
+            match r_reg.get_class() {
+                RegClass::I64 => {
+                    insts.push(Inst::push64(RMI::reg(r_reg.to_reg())));
+                    callee_saved_used += 8;
+                }
+                _ => unimplemented!(),
+            }
+        }
+
+        let mut total_stacksize = self.stack_slots_size + 8 * self.num_spill_slots.unwrap();
+        if self.call_conv.extends_baldrdash() {
+            // Baldrdash expects the stack to expect at least the number of words set in
```

Comment:
```suggestion
            // Baldrdash expects the stack to take at least the number of words set in
```

Or maybe I'm misreading :)